cpp/include/nvtx3/nvtx3.hpp

```diff
@@ -1906,9 +1936,28 @@ struct range_handle {
   constexpr value_type get_value() const noexcept { return _range_id; }
 
  private:
-  value_type _range_id{};  ///< The underlying NVTX range id
+  value_type _range_id{null_handle};  ///< The underlying NVTX range id
 };
 
+/**
+ * @brief Compares two range_handles for equality
+ * 
+ * @param lhs The first range_handle to compare
+ * @param rhs The second range_handle to compare
+ */
+constexpr bool operator==(range_handle lhs, range_handle rhs)
+{
+  return lhs.get_value() == rhs.get_value();
+}
+
+/**
+ * @brief Compares two range_handles for inequality
+ * 
+ * @param lhs The first range_handle to compare
+ * @param rhs The second range_handle to compare
+ */
+constexpr bool operator!=(range_handle lhs, range_handle rhs) { return !(lhs == rhs); }
```

Comment:
Are you suggesting the `operator==/!=` need to be marked `noexcept` in addition to `constexpr`?

Reply:
I suggest to either mark all `constexpr` functions `noexcept`.  Or to never mark `constexpr` functions `noexcept` (since they are by default to my knowledge).  But not to do a mix of both.

I see in your updated change you added `noexcept` to those functions.

Reply:
`constexpr` doesn't imply `noexcept` as certain invocations of a `constexpr` function are allowed to throw as long as there's at least one that doesn't: https://stackoverflow.com/questions/34555793/does-constexpr-imply-noexcept

But otherwise I agree, it was an oversight to not add `noexcept` in all the places you identified. I've fixed that.